datumaro/util/__init__.py

```diff
@@ -14,8 +15,12 @@
 
 str_to_bool = attrs.converters.to_bool
 
+T = TypeVar("T")
 
-def find(iterable, pred=lambda x: True, default=None):
+
+def find(
+    iterable: Iterable[T], pred: Callable[[T], bool] = lambda x: True, *, default: T = None
+) -> T:
```

Comment:
I think more properly `default` should have type `U` (a different type variable) and the return value should have type `Union[T, U]`. Otherwise, a default of `None` would not be type-compatible.

Reply:
Agree, done.